.github/workflows/cd-pipeline.yaml

```diff
@@ -86,6 +86,17 @@ jobs:
             # Container folder doesn't exists we are assuming a deprecation
             echo "::set-output name=result::skip"
           fi
+      - id: get-registry-credentials
+        name: Get staging marketplace's registry credentials
+        run: |
+          csp_auth_token=$(curl -s -H 'Content-Type: application/x-www-form-urlencoded' -X POST -d "grant_type=refresh_token&api_token=${{ secrets.STAGING_MARKETPLACE_API_TOKEN }}" https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/authorize | jq -r .access_token)
+          repo_info=$(curl -s -X POST -H "Content-Type: application/json" -H "csp-auth-token:$csp_auth_token" -d '{"withCredentials":true, "storageType":"OCI"}' https://gtwstg.market.csp.vmware.com/api/v1/repositories/transient)
+          marketplace_user=$(echo "$repo_info" | jq -r .response.repodetails.username)
+          marketplace_passwd=$(echo "$repo_info" | jq -r .response.repodetails.token)
+          echo "::add-mask::${marketplace_user}"
+          echo "::add-mask::${marketplace_passwd}"
+          echo "::set-output name=marketplace_user::${marketplace_user}"
+          echo "::set-output name=marketplace_passwd::${marketplace_passwd}"
```

Comment:
By any chance, are these values visible somewhere in the UI?

Reply:
Nevermind, I've just seen the `add-mask` part, which I assume is used for that purpose.

Reply:
No, they aren't. By default, the `set-output` function does show the variable's value in the logs but the `add-mask` one makes the variable be treated as a secret (it shows *** in the UI). You can see an example of it in the [linked fork run](https://github.com/FraPazGal/containers/runs/8245071416?check_suite_focus=true):

```
::add-mask::***
::add-mask::***
::set-output name=marketplace_user::***
##[debug]steps.get-registry-credentials.outputs.marketplace_user='***'
::set-output name=marketplace_passwd::***
##[debug]steps.get-registry-credentials.outputs.marketplace_passwd='***'
##[debug]Finishing: Get staging marketplace's registry credentials
...
Substituting variable VIB_ENV_REGISTRY_URL in golang/vib-publish.json
Substituting variable VIB_ENV_REGISTRY_USERNAME in golang/vib-publish.json
Substituting variable VIB_ENV_REGISTRY_PASSWORD in golang/vib-publish.json
```